### PR TITLE
Improvements to assets transformations

### PR DESCRIPTION
**Pull request by osdnk**
UPDATE RAINBOW SCRIPTS BEFORE TESTING TO AVOID CACHING ISSUES

This PR creates a bunch of improvements to our Images patterns.
I don't include a ticket for this cause all of them are results of observations we have from https://www.notion.so/rainbowdotme/Jarich-Micha-imgix-bullshit-cccbcbb14a074336a42593c4419ae1d9

and I was just fixing them when noticing what is wrong. 

I made a bunch of improvements:

1. `ImgIxImage` is now `TransformationImage`. This makes much more sense because we also use a bit of cloudinary logic there when we do something related to assets (introduced in https://github.com/rainbow-me/rainbow/pull/3490). 
2. When no options are applied (and the domain is considered safe), we skip signing and use the URL as is. If we don't do anything, we can just skip it. We can do this on a higher level, but I believe the change above is actually hiding a bit of abstraction, so we might also hide this. 
4. in `src/components/unique-token/UniqueTokenImage.js` we remove the transformation to `png` in `ImageTile` that is already accepting `lowResUrl` and `imageUrl` that is `png` (via `svgtoPngIfNeeded` used in `src/parsers/uniqueTokens.js:52`)
5. We don't use signing and imgix stuff if we need only to change the size of `googleusercontent` images. This is enough to just change the size in the URL. 

This should reduce our ImgIx usage by about 40-60% 

no changes observable 

## Dev checklist for QA: what to test
- firstly run the app in develop, then switch to this. This is important to test if the migration works correctly, because we need to be sure that imgix cache will not return old data 

Test all NFTs. Particularly focus on the things you see on the wallet screen and in an expanded state. `Dame.eth` is a good wallet. 
test:
- Normal NFTs
- Poaps
- Uniswap positions (animated SVGs)
- shields (non-animated svgs)

### CODE REVIEW

File: src/handlers/googleUserContent.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+export function adjustSize(
```
Comment (skylarbarrera) [inline src/handlers/googleUserContent.ts:1]: nit but should this be in this file? may be time for a imageUtils

File: src/handlers/imgix.ts [hunk 2]
```diff
@@ -192,11 +202,31 @@ const isPossibleToSignUri = (externalImageUri: string | undefined): boolean => {
...
   externalImageUri: string | undefined,
   options?: ImgOptions,
   skipCaching: boolean = false
 ): string | undefined => {
+  // If no options applied we don't need anything
+  const isTransformationRedundant =
+    !options?.w &&
+    !options?.h &&
+    !options?.fm &&
+    TRUSTED_URLS[externalImageUri?.split('/')[2] ?? ''];
```
Comment (skylarbarrera) [inline src/handlers/imgix.ts:224]: how do we know its redundant for all of out trust urls?

**Review by skylarbarrera — APPROVED** [review]
this looks good post QA but I am team `RainbowImage`

**Review by tchayen — CHANGES_REQUESTED** [review]
`FastCoinIcon.tsx` has the old `ImgIxImage` and crashes the bundler (I changed it locally to be able to continue).

I am not sure if it is directly related to the change, but 50% of times I start the app on Android, I have many coin icons that don't load (USDC, WETH for example) and most NFT collection icons don't show up at all. I also tried waiting couple minutes but they didn't load over time.

photo_2022-06-22 12 10 42

Other than those two problems (or one, as I can't prove the other one is also caused by the changes) looks great 😄.

**Comment by osdnk** [comment]
@tchayen the first thing is because you had not updated the internals.
The second is because of the ImgIx cache that I migrated now.

File: src/model/migrations.ts [hunk 3]
```diff
@@ -680,6 +680,19 @@ export default async function runMigrations() {
 
   migrations.push(v17);
 
+  /*
+   *************** Migration v18 ******************
+   * Removes old MMKV cache
```
Comment (tchayen) [inline src/model/migrations.ts:685]:
```suggestion
   * Removes old MMKV cache for Imgix images
```

I'd add a mention that this only touches image part.

File: src/handlers/googleUserContent.ts (hunk 1 above)
Comment (osdnk) [reply src/handlers/googleUserContent.ts:1]: those are related to the specific format offered by googleUserContent

**Review by jkadamczyk — APPROVED** [review]

**Comment by jkadamczyk** [comment]
Tried with non-empty wallet on Pixel 6 and Xiaomi Redmi, worked for me the images were there

**Review by estebanmino — COMMENTED** [review]
went through the app in a samsung s9 and iphone 11 sim, looking good!
one thing that i see but wondering if is already in develop
this is android
image

which is similar to this issue 
image

**Comment by osdnk** [comment]
Hey @estebanmino we like it or not, this is what i see on develop as well. But thanks for pointing it out!

**Comment by BrodyHughes** [comment]
Messaging Michal about a couple things on this.

File: src/handlers/googleUserContent.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    return `${base}=${extension}`;
+  }
+
+  const url = extension ? `${base}=${extension}` : orinalUrl;
+
+  return url;
+}
+
+export function isGoogleUserContent(url: string | undefined): boolean {
+  return url?.startsWith('https://lh3.googleusercontent.com') ?? false;
```
[the remark this answers is not in the document]
Comment (osdnk) [reply src/handlers/googleUserContent.ts:28]: This is a false positive here. We don't check this for security reasons and the forgery here is not a vulnerability.

File: src/model/migrations.ts [hunk 5]
```diff
@@ -679,6 +679,19 @@ export default async function runMigrations() {
...
   migrations.push(v17);
 
+  /*
+   *************** Migration v18 ******************
+   * Removes old MMKV cache for Imgix images
+   */
+  const v18 = async () => {
+    const imgixMMKV = new MMKV({
+      id: 'IMGIX_CACHE',
+    });
```
Comment (brunobar79) [inline src/model/migrations.ts:689]: we have this globally everywhere else

**Review by brunobar79 — APPROVED** [review]

**Comment by estrattonbailey** [comment]
@osdnk is this PR worth saving? Or should we close, re-scope and prioritize it for future work?

**Comment by jkadamczyk** [comment]
I'm closing this, since we lack too much context about all this. If we think we should still pursue merging it and getting it over the finish line we can re-open this PR and rebase it

**Pull request closed without merging**